plugins/com.google.cloud.tools.eclipse.util/src/com/google/cloud/tools/eclipse/util/MappedNamespaceContext.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.util;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+import javax.xml.XMLConstants;
+import javax.xml.namespace.NamespaceContext;
+
+public class MappedNamespaceContext implements NamespaceContext {
+
+  private final Map<String, String> prefixMapping = new HashMap<>();
+
+  public MappedNamespaceContext(String prefix, String namespaceUri) {
+    checkMapping(prefix, namespaceUri);
+    prefixMapping.put("xml", XMLConstants.XML_NS_URI);
+    prefixMapping.put(prefix, namespaceUri);
+  }
+
+  public MappedNamespaceContext(Map<String, String> prefixMapping) {
+    for (Entry<String, String> entry : prefixMapping.entrySet()) {
+      checkMapping(entry.getKey(), entry.getValue());
+    }
+    this.prefixMapping.put("xml", XMLConstants.XML_NS_URI);
+    this.prefixMapping.putAll(prefixMapping);
+  }
+
+  private static void checkMapping(String prefix, String namespaceUri) {
+    if (prefix == null) {
+      throw new IllegalArgumentException("Prefix can't be null");
+    } else if (prefix.equals("xml")) {
+      throw new IllegalArgumentException("Cannot redefine the 'xml' prefix");
```

Comment:
This is only true if the namespace URI is not http://www.w3.org/XML/1998/namespace